src/commands/dev/index.js

```diff
@@ -35,47 +34,53 @@ const startStaticServer = async ({ settings, log }) => {
   log(`\n${NETLIFYDEVLOG} Server listening to`, settings.frameworkPort)
 }
 
+const isNonExistingCommandError = (error) => {
+  // `ENOENT` is only returned for non Windows systems
+  // See https://github.com/sindresorhus/execa/pull/447
+  if (error.code === 'ENOENT') {
+    return true
+  }
+
+  // this only works on English versions of Windows
+  return (
+    typeof error.message === 'string' && error.message.includes('is not recognized as an internal or external command')
+  )
+}
+
 const startFrameworkServer = async function ({ settings, log, exit }) {
   if (settings.noCmd) {
     return await startStaticServer({ settings, log })
   }
 
   log(`${NETLIFYDEVLOG} Starting Netlify Dev with ${settings.framework || 'custom config'}`)
-  const commandBin = await which(settings.command).catch((error) => {
-    if (error.code === 'ENOENT') {
-      throw new Error(
-        `"${settings.command}" could not be found in your PATH. Please make sure that "${settings.command}" is installed and available in your PATH`,
+
+  const command = `${settings.command} ${settings.args.join(' ')}`
+  // we use reject=false to avoid rejecting synchronously when the command doesn't exist
+  // we can't try->await->catch since we don't want to block on the framework server which
+  // is a long running process
+  const frameworkProcess = execa.command(command, { preferLocal: true, reject: false })
+  frameworkProcess.stdout.pipe(stripAnsiCc.stream()).pipe(process.stdout)
+  frameworkProcess.stderr.pipe(stripAnsiCc.stream()).pipe(process.stderr)
+  process.stdin.pipe(frameworkProcess.stdin)
+
+  // eslint-disable-next-line promise/catch-or-return,promise/prefer-await-to-then
+  frameworkProcess.then(async () => {
+    const result = await frameworkProcess
+    const { exitCode = 0 } = result
+    // eslint-disable-next-line promise/always-return
+    if (result instanceof Error && isNonExistingCommandError(result)) {
+      log(NETLIFYDEVERR, `Failed launching framework server. Please verify ${chalk.magenta(`'${command}' exists`)}`)
+    } else {
+      log(
+        exitCode > 0 ? NETLIFYDEVERR : NETLIFYDEVWARN,
+        `"${command}" exited with code ${exitCode}. Shutting down Netlify Dev server`,
       )
     }
-    throw error
-  })
-  const ps = childProcess.spawn(commandBin, settings.args, {
-    env: { ...process.env, ...settings.env, FORCE_COLOR: 'true' },
-    stdio: 'pipe',
-  })
-
-  ps.stdout.pipe(stripAnsiCc.stream()).pipe(process.stdout)
-  ps.stderr.pipe(stripAnsiCc.stream()).pipe(process.stderr)
-
-  process.stdin.pipe(process.stdin)
-
-  const handleProcessExit = function (code) {
-    log(
-      code > 0 ? NETLIFYDEVERR : NETLIFYDEVWARN,
-      `"${[settings.command, ...settings.args].join(' ')}" exited with code ${code}. Shutting down Netlify Dev server`,
-    )
     process.exit(1)
```

Comment:
We should avoid `process.exit` in general as it kills the CLI process abruptly.
For example it prevents [this code](https://github.com/netlify/cli/blob/c883b6e76aca955b3c4d04475ec96c61993827c9/src/utils/telemetry/tracked-command.js#L38) from running.

I'll handle it in another PR